pkg/agent/agent.go

```diff
@@ -535,11 +525,27 @@ func (a *Agent) updateGroupConfig(obj runtime.Object) {
 		return
 	}
 
+	a.updateConfig(cfg)
+}
+
+func (a *Agent) updateConfig(cfg metav1.Object) {
 	if cfg == nil {
 		log.Warnf("node (%s) has no effective configuration", a.nodeName)
 		return
 	}
 
+	if v, ok := cfg.(cfgapi.Validator); ok {
+		err := v.Validate()
+
+		if err != nil {
```

Comment:
one possiblity to streamline:


```suggestion
		if err := v.Validate(); err != nil {
```

Reply:
Updated.